SimpleQueryStringBuilderTests fails on 6.x

Recent failures: https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+6.x+intake/239/ and https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+6.x+multijob-unix-compatibility/os=oraclelinux/71/console .

```
11:11:34 FAILURE 0.01s J4 | SimpleQueryStringBuilderTests.testUnmappedFieldNoTokenWithAndOperator <<< FAILURES!
11:11:34    > Throwable #1: java.lang.AssertionError: expected:<+mapped_string:first +mapped_string:second> but was:<MatchNoDocsQuery("empty string passed to query parser")>
11:11:34    > 	at __randomizedtesting.SeedInfo.seed([9937D8E57DE82F56:A3EFDB46BCC88FD2]:0)
11:11:34    > 	at org.elasticsearch.index.query.SimpleQueryStringBuilderTests.testUnmappedFieldNoTokenWithAndOperator(SimpleQueryStringBuilderTests.java:727)
11:11:34    > 	at java.lang.Thread.run(Thread.java:748)
```
```
FAILURE 1.79s | SimpleQueryStringBuilderTests.testToQuery <<< FAILURES!
   > Throwable #1: java.lang.AssertionError: 
   > Expected: an instance of org.apache.lucene.search.DisjunctionMaxQuery
   >      but: <MatchNoDocsQuery("empty string passed to query parser")> is a org.apache.lucene.search.MatchNoDocsQuery
   >    at __randomizedtesting.SeedInfo.seed([C7621377DC53D26C:30991149ADD01786]:0)
   >    at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
   >    at org.elasticsearch.index.query.SimpleQueryStringBuilderTests.doAssertLuceneQuery(SimpleQueryStringBuilderTests.java:278)
   >    at org.elasticsearch.index.query.SimpleQueryStringBuilderTests.doAssertLuceneQuery(SimpleQueryStringBuilderTests.java:67)
   >    at org.elasticsearch.test.AbstractQueryTestCase.assertLuceneQuery(AbstractQueryTestCase.java:522)
   >    at org.elasticsearch.test.AbstractQueryTestCase.testToQuery(AbstractQueryTestCase.java:436)
   >    at java.lang.Thread.run(Thread.java:748)
```

Reproduce lines (they both reproduce for me):  

```
./gradlew :server:test -Dtests.seed=9937D8E57DE82F56 -Dtests.class=org.elasticsearch.index.query.SimpleQueryStringBuilderTests -Dtests.method="testUnmappedFieldNoTokenWithAndOperator" -Dtests.security.manager=true -Dtests.locale=sr-RS -Dtests.timezone=America/Argentina/Salta -Dcompiler.java=11 -Druntime.java=8
```

```
./gradlew :server:test \
  -Dtests.seed=C7621377DC53D26C \
  -Dtests.class=org.elasticsearch.index.query.SimpleQueryStringBuilderTests \
  -Dtests.method="testToQuery" \
  -Dtests.security.manager=true \
  -Dtests.locale=fr-FR \
  -Dtests.timezone=Asia/Gaza \
  -Dcompiler.java=11 \
  -Druntime.java=8
```

I suspect this is due to https://github.com/elastic/elasticsearch/commit/92331fdb6b333b65077418ecf4ec8e701b23553f .

--- a/server/src/test/java/org/elasticsearch/index/query/SimpleQueryStringBuilderTests.java
+++ b/server/src/test/java/org/elasticsearch/index/query/SimpleQueryStringBuilderTests.java
@@ -271,8 +271,7 @@ public class SimpleQueryStringBuilderTests extends AbstractQueryTestCase<SimpleQ
     @Override
     protected void doAssertLuceneQuery(SimpleQueryStringBuilder queryBuilder, Query query, SearchContext context) throws IOException {
         assertThat(query, notNullValue());
-
-        if (queryBuilder.value().isEmpty()) {
+        if (queryBuilder.value().isEmpty() || getCurrentTypes().length == 0) {
             assertThat(query, instanceOf(MatchNoDocsQuery.class));
         } else if (queryBuilder.fields().size() > 1) {
             assertThat(query, instanceOf(DisjunctionMaxQuery.class));
@@ -707,20 +706,14 @@ public class SimpleQueryStringBuilderTests extends AbstractQueryTestCase<SimpleQ
         assertEquals(expected, query);
     }
 
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/35773")
-    @Override
-    public void testToQuery() throws IOException {
-        super.testToQuery();
-    }
-
     /**
      * Test for behavior reported in https://github.com/elastic/elasticsearch/issues/34708
      * Unmapped field can lead to MatchNoDocsQuerys in disjunction queries. If tokens are eliminated (e.g. because
      * the tokenizer removed them as punctuation) on regular fields, this can leave only MatchNoDocsQuerys in the
      * disjunction clause. Instead those disjunctions should be eliminated completely.
      */
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/35773")
     public void testUnmappedFieldNoTokenWithAndOperator() throws IOException {
+        assumeTrue("test runs only when at least a type is registered", getCurrentTypes().length > 0);
         Query query = new SimpleQueryStringBuilder("first & second")
                 .field(STRING_FIELD_NAME)
                 .field("unmapped")

